core/optaplanner-core-impl/src/main/java/org/optaplanner/core/impl/heuristic/selector/move/generic/list/KOptDescriptor.java

```diff
@@ -0,0 +1,271 @@
+package org.optaplanner.core.impl.heuristic.selector.move.generic.list;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import org.optaplanner.core.api.function.TriPredicate;
+import org.optaplanner.core.impl.domain.variable.descriptor.ListVariableDescriptor;
+import org.optaplanner.core.impl.domain.variable.index.IndexVariableSupply;
+import org.optaplanner.core.impl.util.Pair;
+
+public final class KOptDescriptor<Solution_> {
```

Comment:
Pretty much every field or method argument in this class needs a better name. Also, `camelCase` needs to be followed consistently.

But I'm assuming you're aware of that and were going to solve it before this PR is ready for review.